ui/views.py

```diff
@@ -63,7 +64,10 @@ class VideoDetail(DetailView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context['form'] = VideoForm(instance=context["object"])
+        video = context["object"]
+        video.status = refresh_status(video)
```

Comment:
This line basically amounts to `video.status = video.status`, so i think you can remove it.